pkg/store/pruning.go

```diff
@@ -0,0 +1,75 @@
+package store
+
+import (
+	"context"
+
+	ds "github.com/ipfs/go-datastore"
+
+	"github.com/rollkit/rollkit/pkg/config"
+	"github.com/rollkit/rollkit/types"
+)
+
+type DefaultPruningStore struct {
+	Store
+
+	Config config.PruningConfig
+}
+
+var _ PruningStore = &DefaultPruningStore{}
+
+// NewDefaultPruningStore returns default pruning store.
+func NewDefaultPruningStore(ds ds.Batching, config config.PruningConfig) PruningStore {
+	return &DefaultPruningStore{
+		Store:  &DefaultStore{db: ds},
+		Config: config,
+	}
+}
+
+// SaveBlockData adds block header and data to the store along with corresponding signature.
+// It also prunes the block data if needed.
+func (s *DefaultPruningStore) SaveBlockData(ctx context.Context, header *types.SignedHeader, data *types.Data, signature *types.Signature) error {
+	if err := s.PruneBlockData(ctx); err != nil {
```

Comment:
The method `SaveBlockData` does not give any indication of pruning the store. Instead, I would suggest to make this logic explicit and not piggy back on an existing process. `SaveBlockData` may be called multiple times in a workflow.